lib/Recipe.js

```diff
@@ -248,6 +248,8 @@ class Recipe {
             } else {
                 return callback();
             }
+        } else if (this.isBufferSrc && this.output) {
+            fs.writeFileSync(this.output, this.outStream.toBuffer());
```

Comment:
Thanks for debugging! Maybe we can clear this a little bit like this?
```javascript
if (this.isBufferSrc && this.output) {
    fs.writeFileSync(this.output, this.outStream.toBuffer());
}

if (callback) {
    if (this.output) {
        return callback(this.output);
    } else if (this.outStream && this.outStream.toBuffer) {
        return callback(this.outStream.toBuffer());
    } else {
        return callback();
    }
}
```

Reply:
Yes, that is much better! I'll fix and create a new pull with one commit so as not to pollute the git log.